lib/enquirer.js

```diff
@@ -0,0 +1,258 @@
+import * as utils from './utils.js';
+import { equals, isTruthy } from './assert.js';
+
+import createPrompt from './prompt.js';
+import createEnquirerTypes from './types/index.js';
+import createEnquirerPrompts from './prompts/index.js';
+
+export default function createEnquirer(platformShims) {
+  const enquirerTypes = createEnquirerTypes(platformShims);
+  const enquirerPrompts = createEnquirerPrompts(platformShims);
+
+  /**
+   * Create an instance of `Enquirer`.
+   *
+   * ```js
+   * const Enquirer = require('enquirer');
+   * const enquirer = new Enquirer();
+   * ```
+   * @name Enquirer
+   * @param {Object} `options` (optional) Options to use with all prompts.
+   * @param {Object} `answers` (optional) Answers object to initialize with.
+   * @api public
+   */
+
+  class Enquirer extends platformShims.EventEmitter {
+    constructor(options, answers) {
+      super();
+      this.options = utils.merge({}, options);
+      this.answers = { ...answers };
+    }
+
+    /**
+     * Register a custom prompt type.
+     *
+     * ```js
+     * const Enquirer = require('enquirer');
+     * const enquirer = new Enquirer();
+     * enquirer.register('customType', require('./custom-prompt'));
+     * ```
+     * @name register()
+     * @param {String} `type`
+     * @param {Function|Prompt} `fn` `Prompt` class, or a function that returns a `Prompt` class.
+     * @return {Object} Returns the Enquirer instance
+     * @api public
+     */
+
+    register(type, fn) {
+      if (utils.isObject(type)) {
+        for (let key of Object.keys(type)) this.register(key, type[key]);
+        return this;
+      }
+      equals(typeof fn, 'function', 'expected a function');
+
+      function isPrompt(object) {
+        if (!object) { return false; }
+        return object.name === 'Prompt' || isPrompt(object.__proto__);
+      }
+
+      const name = type.toLowerCase();
+      this.prompts[name] = isPrompt(fn) ? fn : fn(this.Prompt, this);
+      return this;
```

Comment:
That part's tough. Now that the classes are built on-demand with platform shims, the previous check: `fn instanceof Prompt` does not work anymore. 

The method here is a hacky way of trying to stay as close to the previous behavior as possible but it's not clean. 

I also tried:

```js
this.prompts[name] = fn.__proto__.name ? fn : fn(this.Prompt, this);
``` 

Which works but seems as hacky as the previous version  and doesn't do exactly the same thing (only checks whether the argument is a constructor, not that is extends Prompt). Given the fact that constructors that didn't extend Prompt already caused issues (because they were called without `new`) I think the solution above is almost acceptable.